hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -1488,15 +1478,15 @@ public long[] findUnknownExecutions(long[] executionIds) {
         }).toArray();
     }
 
-    private void scheduleJobTimeout(final long jobId, final long timeout) {
+    void scheduleJobTimeout(final long jobId, final long timeout) {
```

Comment:
timeout could be ignored if it was very short and the job took long to start (for example there are a few IMap operation in between). we could fix this as well

Reply:
in such case job should fail/be cancelled immediately

Reply:
Yes, theoretically, it can be "fixed," but specifying a timeout which is less than a second is a misuse.

Reply:
It was a problem because in previous PR version you passed negative timeout here, but that is no more possible.